vslib/src/SwitchStateBase.cpp

```diff
@@ -2593,28 +2591,16 @@ sai_status_t SwitchStateBase::set_system_port_list()
     return set(SAI_OBJECT_TYPE_SWITCH, m_switch_id, &attr);
 }
 
-sai_status_t SwitchStateBase::set_voq_switch_attributes(
-        _In_ int32_t voq_switch_id,
-        _In_ uint32_t voq_max_cores)
+sai_status_t SwitchStateBase::refresh_system_port_list(
+        _In_ const sai_attr_metadata_t *meta)
 {
     SWSS_LOG_ENTER();
 
-    SWSS_LOG_INFO("set voq switch attributes");
-
-    sai_attribute_t attr;
-
-    attr.id = SAI_SWITCH_ATTR_TYPE;
-    attr.value.s32 = SAI_SWITCH_TYPE_VOQ;
-
-    CHECK_STATUS(set(SAI_OBJECT_TYPE_SWITCH, m_switch_id, &attr));
-
-    attr.id = SAI_SWITCH_ATTR_SWITCH_ID;
-    attr.value.u32 = voq_switch_id;
+    // Currently, system ports info are not changing and also since currently the local
+    // port oid for local system ports is retrieved only during portsorch initialization,
+    // no referesh is done
```

Comment:
> local port oid for local system ports is retrieved only during portsorch initialization, no referesh is done

this is actually not true:
https://github.com/Azure/sonic-swss/blob/master/orchagent/portsorch.cpp#L1759
https://github.com/Azure/sonic-swss/blob/master/orchagent/portsorch.cpp#L1778

ports are remove and added inside portsorch, and this is happening after switch is created, so let it may happen that after switch_create you have 32 local ports, and then orchagent will remove some ports and add some ports, and you can end up in total 48 local ports (since for example 100G port was splitted to 4x25G ports) then probably VOQ configuration is not valid any more at this point

Reply:
I meant this with respect to the upcoming PR https://github.com/Azure/sonic-swss/pull/1431. While adding System Ports (already pre-configured as part of switch create) in PortOrch constructor, to map the system ports corresponding to the local ports, the "local port oid" attribute of the system port info is retrieved. This is the only place we will be in need of this until dynamic system port add/removal is handled. 
If this is not clear, I'll remove this statement.

Reply:
i have only a little bit knowledge on orchagent, but im better on sairedis/syncd/vs part, but from recordings what i saw, that ports are dynamically splitted in orchagent, and i remember that since i added unisttests for this and support to VS for that scenario. I also read https://github.com/Azure/SONiC/blob/master/doc/voq/voq_hld.md, but its mostly high level description, where on SAI level is very little description, and bottom page just says dynamic part is future work

I'm pointing this dynamic part here since i have been in similar scenarios previously, and it was on my head to find issues why X is not working, because Y was not included in dynamic breakout scenario, and this VOQ switch system seems to be to be even more complicated in this part when it will come to create/remove ports.

I think also question will be here for this feature be supported on sonic or you just adding support to sairedis/syncd/vs to use in internal (nokia?) projects? I already know some features that other vendors added, but they will not be used publicly in sonic

Reply:
Yes. Dynamic system port handling is future work. Not sure when it will be taken. I agree with you that when the dynamic system ports add/del is taken, it is going to be bit complex. That is one of the reasons why it was proposed to start with simple static pre-configuration and do dynamic system ports add/del later. This VoQ System Ports is part of SONiC on Chassis System development.